### ISSUE TITLE
关于直接下载您处理好的cstnet-tls1.3数据集的疑问

### ISSUE THREAD

**Issue by MiloGame** [linwhitehat/ET-BERT#68]
作者大大您好，我还有一个小问题，就是直接下载您处理好的cstnet-tls1.3用于微调，我发现里面好像没有text_b，只有text_a，如下图所示。
![image](https://github.com/linwhitehat/ET-BERT/assets/125568740/b7143913-0a37-4868-a38d-937cdc901062)
而且您微调的命令
python3 fine-tuning/run_classifier.py --pretrained_model_path models/pre-trained_model.bin \
                                   --vocab_path models/encryptd_vocab.txt \
                                   --train_path datasets/cstnet-tls1.3/packet/train_dataset.tsv \
                                   --dev_path datasets/cstnet-tls1.3/packet/valid_dataset.tsv \
                                   --test_path datasets/cstnet-tls1.3/packet/test_dataset.tsv \
                                   --epochs_num 10 --batch_size 32 --embedding word_pos_seg \
                                   --encoder transformer --mask fully_visible \
                                   --seq_length 128 --learning_rate 2e-5
这行命令中调用的脚本里读入微调数据的函数部分，如下图，里面好像并没有执行句子对的构成，也就和您论文里说进行第二种句子对的训练不吻合，读入的数据只有text_a的部分
<img width="511" alt="image" src="https://github.com/linwhitehat/ET-BERT/assets/125568740/2734bf5f-b176-4e43-872e-0754574f26c8">
此外vocab_path这个参数好像没有用到欸，没有在这个脚本里检索到
<img width="737" alt="image" src="https://github.com/linwhitehat/ET-BERT/assets/125568740/4b9677e5-bd30-4f12-a263-0b4bfecc30ac">
还请大佬能答疑解惑，祝您生活愉快，国庆快乐

**Comment by 1195986182** [comment]
我下载您的CSTNET-TLS1.3数据集，里面包含的是datagram和label的npy文件，这部分直接合并可以得到tsv文件内容吗？

**Comment by linwhitehat** [comment]
> 我下载您的CSTNET-TLS1.3数据集，里面包含的是datagram和label的npy文件，这部分直接合并可以得到tsv文件内容吗？

可以的

**Comment by linwhitehat** [comment]
> 作者大大您好，我还有一个小问题，就是直接下载您处理好的cstnet-tls1.3用于微调，我发现里面好像没有text_b，只有text_a，如下图所示。 ![image](https://private-user-images.githubusercontent.com/125568740/271813637-b7143913-0a37-4868-a38d-937cdc901062.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzMDk4MjIsIm5iZiI6MTcwNTMwOTUyMiwicGF0aCI6Ii8xMjU1Njg3NDAvMjcxODEzNjM3LWI3MTQzOTEzLTBhMzctNDg2OC1hMzhkLTkzN2NkYzkwMTA2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNVQwOTA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNjVmMTA5ZDdjN2FlYzY1MWQ4MjkzMDM5NTA5ZTY2NmI4NWFlZDlhYzhjNGNmZmQxMDA3YTRhNjA3ZDRhOWU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1cqtNVzQ7anSchVP3WozgVKA_yFXfI2pTe5e6XxEugY) 而且您微调的命令 python3 fine-tuning/run_classifier.py --pretrained_model_path models/pre-trained_model.bin --vocab_path models/encryptd_vocab.txt --train_path datasets/cstnet-tls1.3/packet/train_dataset.tsv --dev_path datasets/cstnet-tls1.3/packet/valid_dataset.tsv --test_path datasets/cstnet-tls1.3/packet/test_dataset.tsv --epochs_num 10 --batch_size 32 --embedding word_pos_seg --encoder transformer --mask fully_visible --seq_length 128 --learning_rate 2e-5 这行命令中调用的脚本里读入微调数据的函数部分，如下图，里面好像并没有执行句子对的构成，也就和您论文里说进行第二种句子对的训练不吻合，读入的数据只有text_a的部分 <img alt="image" width="511" src="https://private-user-images.githubusercontent.com/125568740/271813770-2734bf5f-b176-4e43-872e-0754574f26c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzMDk4MjIsIm5iZiI6MTcwNTMwOTUyMiwicGF0aCI6Ii8xMjU1Njg3NDAvMjcxODEzNzcwLTI3MzRiZjVmLWIxNzYtNGU0My04NzJlLTA3NTQ1NzRmMjZjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNVQwOTA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTZkM2U3NDgyMjk2MzFkNzA0N2M3ZmIzZTI5ZGYzMTQ0ZDIyZDhmZTM5Y2UyODdhMGYzMzk2Y2M0ODRhMWJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZP9ikO7CK5zLrdTFc5ZC1W0gG0FPT5Szazc2ZY-m_Uc"> 此外vocab_path这个参数好像没有用到欸，没有在这个脚本里检索到 <img alt="image" width="737" src="https://private-user-images.githubusercontent.com/125568740/271813739-4b9677e5-bd30-4f12-a263-0b4bfecc30ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzMDk4MjIsIm5iZiI6MTcwNTMwOTUyMiwicGF0aCI6Ii8xMjU1Njg3NDAvMjcxODEzNzM5LTRiOTY3N2U1LWJkMzAtNGYxMi1hMjYzLTBiNGJmZWNjMzBhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNVQwOTA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjRmOTMzODdkYTA5NWZiNmUyY2JiZTYyOWIxNWQ1OTg2ODY4Nzc5MTkwZDJiN2ZlNWQ2ZTYxNzg5ZTE3ZGQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.g7QG5ouKCLGlhIhcoewPCKxbs6NClhk12Zz7qE_0eZI"> 还请大佬能答疑解惑，祝您生活愉快，国庆快乐

你提及的部分是预训练的数据处理，在微调阶段并不需要特意构建text_a和text_b，而是考虑应用任务的需要去处理。

**Comment by huxuan-sda** [comment]
> 我下载您的CSTNET-TLS1.3数据集，里面包含的是datagram和label的npy文件，这部分直接合并可以得到tsv文件内容吗？

请问怎么合并呢，求

**Comment by Chenxin2000** [comment]
我想知道这样合并的tsv是流级别的还是包级别的
